src/FakeItEasy/Configuration/PropertySetterConfiguration.cs

```diff
@@ -0,0 +1,92 @@
+namespace FakeItEasy.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using FakeItEasy.Core;
+
+    internal class PropertySetterConfiguration<TValue>
+        : IPropertySetterConfiguration<TValue>
+    {
+        private readonly MethodCallExpression setterExpression;
+
+        private readonly Func<LambdaExpression, IVoidArgumentValidationConfiguration>
+            voidArgumentValidationConfigurationFactory;
+
+        public PropertySetterConfiguration(
+            MethodCallExpression setterExpression,
+            Func<LambdaExpression, IVoidArgumentValidationConfiguration> voidArgumentValidationConfigurationFactory)
+        {
+            this.setterExpression = setterExpression;
+            this.voidArgumentValidationConfigurationFactory = voidArgumentValidationConfigurationFactory;
+        }
+
+        public IVoidArgumentValidationConfiguration To(TValue value)
+        {
+            Expression<Func<TValue>> valueExpression = () => value;
+            var newSetterExpression = this.CreateSetterExpressionWithNewValue(valueExpression);
+            return this.CreateArgumentValidationConfiguration(newSetterExpression);
+        }
+
+        public IAfterCallSpecifiedConfiguration Throws(Func<IFakeObjectCall, Exception> exceptionFactory)
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .Throws(exceptionFactory);
+        }
+
+        public IVoidConfiguration Invokes(Action<IFakeObjectCall> action)
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .Invokes(action);
+        }
+
+        public IAfterCallSpecifiedConfiguration CallsBaseMethod()
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .CallsBaseMethod();
+        }
+
+        public IAfterCallSpecifiedConfiguration AssignsOutAndRefParametersLazily(

```

Comment:
> out and ref aren't supported even for indexers, it seems?

No, they're not.

> We could throw an invalid operation (we may anyhow).

We could, but it would be better if we could prevent it at compile time.

> Or would you rather we made up new interfaces for the fluent API to avoid confusion?

It's unfortunate, but yes, I think it would be the right way to handle that...

Maybe we could change `IVoidConfiguration` so that it no longer inherits `IOutAndRefParametersConfiguration`, and introduce a new `IVoidWithOutAndRefParametersConfiguration`? (it would be a breaking change, but unlikely to affect many people). It would probably be the easiest route; creating a completely new interface would require an adapter.